logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventMapper.kt

```diff
@@ -185,9 +187,32 @@ class EventMapper(
         eventContentDTO: EventContentDTO.UserProperty.PropertiesSetDTO,
         transient: Boolean
     ): Event {
-        return when (val fieldKeyValue = eventContentDTO.value) {
-            is EventContentDTO.FieldKeyNumberValue -> ReadReceiptModeSet(id, transient, fieldKeyValue.value == 1)
-            is EventContentDTO.FieldUnknownValue -> unknown(
+        val fieldKeyValue = eventContentDTO.value
+        val key = eventContentDTO.key
+        return when {
+            fieldKeyValue is EventContentDTO.FieldKeyNumberValue -> {
+                when (key) {
+                    WIRE_RECEIPT_MODE.key -> ReadReceiptModeSet(
+                        id,
+                        transient,
+                        fieldKeyValue.value == 1
+                    )
+
+                    WIRE_TYPING_INDICATOR_MODE.key -> TypingIndicatorModeSet(
+                        id,
+                        transient,
+                        false
```

Comment:
question: shouldn't the `fieldKeyValue.value` be compared here as well instead of the hardcoded `false`? 
I found this: https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/29065437/Use+case+Typing+indicator#Syncing-typing-indicator-preference-for-each-account-across-the-devices
I think that only `0` should be treated as disabled, not any value, right? 🤔

Reply:
I agree this is indeed a confusing way to handle the flag enable/disable value. The thing is that due to the nature of the feature itself (opt-out / enabled by default) the Backend is only sending:

1. A `user.properties-set` event with key `WIRE_TYPING_INDICATOR_MODE` and value `0` when the flag gets disabled on other clients
2. And a `user.properties-delete` event with key `WIRE_TYPING_INDICATOR_MODE` when enabling it on other clients

This means that `updateUserProperties()` will never be called with `WIRE_TYPING_INDICATOR_MODE` key when enabling the feature. But if you think changing it to `fieldKeyValue.value != 0` is less future error prone we can also leave the comparison.